.golangci.yaml

```diff
@@ -1,55 +1,185 @@
+  varcheck:
+    exported-fields: true
 
 linters:
-  disable-all: true
   enable:
-    - errcheck
-    - gofmt
-    - goimports
-    - gosimple
-    - ineffassign
-    - misspell
-    - staticcheck
-    - unconvert
-    - unused
-    - govet
-    - gocyclo
+    - asasalint
+    - asciicheck
+    - bidichk
+    - bodyclose
+    # TODO(containedctx): revisit!
+    #- containedctx
+    # TODO(contextcheck): revisit
+    #- contextcheck
+    - dogsled
     - dupl
-    - errchkjson
+    - durationcheck
+    - errname
+    # TODO(errorlint): revisit
+    #- errorlint
     - exhaustive
-    - gocognit
-    - goconst
+    - exportloopref
+    # TODO(forbidigo): revisit
+    #- forbidigo
+    # TODO(forcetypeassert): revisit this one! beware fragile asserts in this codebase
+    #- forcetypeassert
+    # TODO(goconst): revisit
+    #- goconst
+    # TODO(gocritic): revisit
+    #- gocritic
+    - gocyclo
+    # TODO(godot): revisit
+    #- godot
+    # TODO(gofumpt): revisit
+    #- gofumpt
+    # TODO(goimports): revisit
+    #- goimports
+    - goprintffuncname
+    # TODO(gosec): revisit; consequences of revoking non-owner file permissions?
+    #- gosec
+    # TODO(interfacebloat): revisit in a followup; will require a breaking API change
+    #- interfacebloat
+    # TODO(ireturn): revisit in a followup; will require a breaking API change
+    #- ireturn
+    # TODO(lll): revisit in followup; formatting
+    #- lll
+    - misspell
+    - nakedret
+    # TODO(nestif): revisit in a followup; refactor will need more careful review
+    #- nestif
+    # TODO(nilerr): revisit; some tricky inversion of err in tests
+    #- nilerr
+    # TODO(nilnil): revisit
+    #- nilnil
+    # TODO(noctx): revisit in a followup; context lifecycle consequences and/or breaking API change
+    #- noctx
+    - nolintlint
+    # TODO(prealloc): revisit in a followup; some logic around slices are ignoring errors
+    #- prealloc
+    - predeclared
+    - promlinter
+    # TODO(revive): revisit in a followup; extensive changes, some breaking. godoc requirement would be good to introduce
+    #- revive
+    - rowserrcheck
+    - sqlclosecheck
+    # TODO(stylecheck): revisit in a followup; some breaking API changes
+    #- stylecheck
+    # NOTE: removed tagliatelle as it conflicts too much with existing API wireformats
     - tenv
+    # TODO(testpackage): improve open vs closed-box testing in a followup
+    #- testpackage
+    - thelper
+    - tparallel
+    - unconvert
+    # TODO(unparam): revisit
+    #- unparam
+    - usestdlibvars
+    - wastedassign
+    - whitespace
+    # TODO(wrapcheck): wrap errors in a followup
+    #- wrapcheck
 
-linters-settings:
-  goimports:
-    local-prefixes: github.com/snyk/snyk-ls
-  exhaustive:
-    default-signifies-exhaustive: true
-
-output:
-  format: tab
-
-run:
-  deadline: 2m
-  skip-dirs:
-    - "docs"
-    - "licenses"
-    - "pact"
-    - "ampli"
-    - ".bin"
-    - ".github"
-    - ".vscode"
-    - "build"
```

Comment:
question: Why not exclude these directories? We really don't want to lint e.g. code that is pulled in via the license in the licenses folder or generated code from ampli/pact.